packages/runtime/runtime-definitions/src/dataStoreContext.ts

```diff
@@ -137,6 +136,14 @@ export interface IContainerRuntimeBase extends
      * Returns the current audience.
      */
     getAudience(): IAudience;
+
+    /**
+     * Returns the number of messages currently pending acknowledgement from the service.
+     *
+     * Clients that produces high volumes of messages should check this count periodically
+     * and reduce production if the service/network is falling behind.
+     */
+    pendingMessageCount: number;
```

Comment:
I'm not a big fan of continuing to increase our global space (container is equivalent to global / singleton in "traditional" programming). I think if we go with this approach, I'd want to see same property on data store / DDS, such that the gap between all three does not continue to increase. I view them all as one and the same in some future of framework, where they are nodes of the tree holding other nodes like them, and exception is that "root" data store has to implement just a bit more things (today's IRuntime) but does not expose any more capabilities (if possible)